packages/ella/src/dom/index.ts

```diff
@@ -913,3 +914,57 @@ export function forceFocusAndOpenKeyboard(element:HTMLElement, timeout = 0) {
     console.error('Unable to force focus input', err);
   }
 }
+
+/**
+ * This method can be used to get the Device details like browser name, user agent, os name and origin i.e desktop or mobile website.
+ *
+ * @remarks
+ * This method depends on userAgent sniffing and therefore susciptible to spoofing. Avoid detecting browsers in business impacting code
+ *
+ * @example
+ * ```
+ * console.log('Get Device Details - ',getDeviceDetails());
+ * ```
+ */
+export function getDeviceDetails() {
+  try {
+    if (typeof window !== 'undefined' && navigator != null) {
+
+      const browserName = getBrowserName();
+
+      const OSName = getOSName();
+
+      let origin = null;
+      const nAgt = navigator.userAgent;
+
+      //Checking here if any of these matches the given user agents then setting origin to mobile website else setting it to Desktop
+      if (/Android|webOS|iPhone|iPad|iPod|BlackBerry|IEMobile|Opera Mini|Mobile|mobile|CriOS/i.test(nAgt)) {
+        origin = BROWSER_ORIGIN.MOBILE;
+
+      } else {
+        origin = BROWSER_ORIGIN.DESKTOP;
+      }
+
+
+      const finalPayload = {
+        browserName: browserName,
+        userAgent: navigator.userAgent,
+        OSName: OSName,
+        origin: origin
```

Comment:
Can comment what do we mean by origin here. Browser, OS and UA is very clear...... Origin is confusing.

Reply:
Have added it in the comments This method can be used to get the Device details like browser name, user agent, os name and origin i.e desktop or mobile website. Don't think need to add here